src/bidiMapper/domains/input/ActionDispatcher.ts

```diff
@@ -0,0 +1,632 @@
+/**
+ * Copyright 2023 Google LLC.
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CommonDataTypes, Input, Message} from '../../../protocol/protocol.js';
+import {assert} from '../../../utils/assert.js';
+import {BrowsingContextImpl} from '../context/browsingContextImpl.js';
+
+import {ActionOption} from './ActionOption.js';
+import {KeySource, PointerSource, WheelSource} from './InputSource.js';
+import {InputState} from './InputState.js';
+import {KeyToKeyCode} from './USKeyboardLayout.js';
+import {getNormalizedKey, getKeyCode, getKeyLocation} from './keyUtils.js';
+
+/** https://w3c.github.io/webdriver/#dfn-center-point */
+const CALCULATE_IN_VIEW_CENTER_PT_DECL = ((i: Element) => {
+  const t = i.getClientRects()[0] as DOMRect,
+    e = Math.max(0, Math.min(t.x, t.x + t.width)),
+    n = Math.min(window.innerWidth, Math.max(t.x, t.x + t.width)),
+    h = Math.max(0, Math.min(t.y, t.y + t.height)),
+    m = Math.min(window.innerHeight, Math.max(t.y, t.y + t.height));
+  return [e + ((n - e) >> 1), h + ((m - h) >> 1)];
+}).toString();
+
+async function getElementCenter(
```

Comment:
Please make it a private static method

Reply:
What's your reason? Generally you make something static if it's related to the creation, destruction, or modification of the class. It's also done for namespacing. I don't think this fits these criteria.

Reply:
Resolving this for now. Reopen if needed.